code/modules/mob/living/carbon/carbon_defense.dm

```diff
@@ -45,6 +45,8 @@
 	return null
 
 /mob/living/carbon/is_pepper_proof(check_flags = ALL)
+	if(HAS_TRAIT(src, TRAIT_PEPERSPRAY_RESISTANCE))
+		return TRUE
```

Comment:
this should be on living side, not on carbon. make this proc call parent and then put this check on the parent proc. also this expects a ref, so i think you wanna return src